File: app/src/main/java/pl/piotrskiba/angularowo/models/AdOffer.java
package pl.piotrskiba.angularowo.models;

public class AdOffer {

    private final String id;
    private final int points;
    private final String adId;
    private final int timeBreak;
    private final int timeleft;

    public AdOffer(String id, int points, String adId, int timeBreak, int timeleft){
        this.id = id;
        this.points = points;
        this.adId = adId;
        this.timeBreak = timeBreak;
        this.timeleft = timeleft;
    }

    public String getId() {
        return id;
    }

    public int getPoints() {
        return points;
    }

    public String getAdId() {
        return adId;
    }

    public int getTimeBreak() {
        return timeBreak;
    }

    public int getTimeleft() {
        return timeleft;
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/main/offers/model/AdOffer.kt
package pl.piotrskiba.angularowo.main.offers.model

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import pl.piotrskiba.angularowo.R
import pl.piotrskiba.angularowo.domain.offers.model.AdOfferModel
import java.util.Date

data class AdOffer(
    override val id: String,
    val points: Int,
    val adId: String,
    private val timeBreak: Int,
    private val availabilityDate: Date?,
    private val pointsLimitReached: Boolean,
) : Offer(id = id, availabilityDate = availabilityDate, redeemable = !pointsLimitReached) {

    @ColorRes
    val pointsTextColor =
        if (canRedeem) {
            R.color.color_coin
        } else {
            R.color.color_coin_disabled
        }
    val pointsText = points.toString()

    @DrawableRes
    val iconDrawable =
        if (canRedeem) {
            R.drawable.ic_coin
        } else {
            R.drawable.ic_coin_disabled
        }
}

fun AdOfferModel.toUi(pointsLimitReached: Boolean) =
    AdOffer(
        id,
        points,
        adId,
        timeBreak,
        availabilityDate,
        pointsLimitReached,
    )

fun List<AdOfferModel>.toUi(pointsLimitReached: Boolean) =
    map { it.toUi(pointsLimitReached) }
